Album/Album/Model/Album.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  Album.swift
+//  Album
+//
+//  Created by Bradley Yin on 9/2/19.
+//  Copyright © 2019 bradleyyin. All rights reserved.
+//
+
+import Foundation
+
+struct Album: Decodable, Encodable {
+    var artist: String
+    var coverArt: String
+    var genres: [String]
+    let id: String
+    var name: String
+    var songs: [Song]?
+    
+    enum CodingKeys: String, CodingKey {
+        case artist
+        case coverArt
+        case genres
+        case id
+        case name
+        case songs
+        
+        enum CoverArtCodingKeys: String, CodingKey {
+            case url
+        }
+    }
+    init(artist: String, genres: [String], name: String, coverArt: String, songs: [Song]) {
+        self.artist = artist
+        self.coverArt = coverArt
+        self.genres = genres
+        self.id = UUID().uuidString
+        self.name = name
+        self.songs = songs
+    }
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        artist = try container.decode(String.self, forKey: .artist)
+        
+        var coverArtContainer = try container.nestedUnkeyedContainer(forKey: .coverArt)
+        let urlContainer = try coverArtContainer.nestedContainer(keyedBy: CodingKeys.CoverArtCodingKeys.self)
```

Comment:
cover art container is actually an array, so you should loop over it, decoding a new url container through each iteration.

what you have will work when there's only one cover, but there CAN be more than one. I believe it would crash if there was 0 covers too, but I'm not 100% sure on that.